### PR TITLE
Remove usage of flatRawNulls method from IsNullFunction

### PR DESCRIPTION
**Pull request by mbasmanova**
Simplify IsNullFunction implementation and remove usage of
BaseVector::flatRawNulls() method. Add tests for constant input.

See https://github.com/facebookincubator/velox/discussions/2202 for context.

### CODE REVIEW

File: velox/functions/lib/IsNull.cpp [hunk 1]
```diff
@@ -32,47 +32,44 @@ class IsNullFunction : public exec::VectorFunction {
...
-        rows.applyToSelected([&](vector_size_t i) {
-          flatResult->set(i, !bits::isBitNull(rawNulls, i));
-        });
-      }
+      isNull = AlignedBuffer::allocate<bool>(rows.size(), context->pool());
+      memcpy(
+          isNull->asMutable<int64_t>(),
+          arg->rawNulls(),
+          bits::nbytes(rows.end()));
+      bits::negate(isNull->asMutable<char>(), rows.end());
```
Comment (Yuhta) [inline velox/functions/lib/IsNull.cpp:63]: We no longer check the selectivity?

Comment (mbasmanova) [reply velox/functions/lib/IsNull.cpp:63]: It is not needed. Most of the time 'result' doesn't have any data that requires preserving, hence, moveOrCopyResult will simply move the localResult. In rare cases when is_null if used under a switch or if, moveOrCopyResult will copy the data as needed.

Comment (Yuhta) [reply velox/functions/lib/IsNull.cpp:63]: I think in the old version, when selectivity is set and result is reused, we do not allocate new buffer but only overwrite the selected rows in the result.  In the new version we do the allocation and compute the full range regardless of the selectivity.

This case is very rare though, only happens with a low selectivity, negated nullability, and reuse of the result.

Comment (mbasmanova) [reply velox/functions/lib/IsNull.cpp:63]: That's correct.

**Comment by kgpai** [comment]
Thanks for this change, I was wondering if you can add a little more context , as per #2202 , since flatRawNulls is not thread safe, there are other locations also its being used; Wont these have to change too ? What would be the cost of making flatRawNulls thread safe (say a lock ?) .

**Comment by mbasmanova** [comment]
> since flatRawNulls is not thread safe, there are other locations also its being used; Wont these have to change too ? 

Yes, all call sites will need to be changed. I'll be sending separate PRs for each.

**Comment by mbasmanova** [comment]
> What would be the cost of making flatRawNulls thread safe (say a lock ?) 

Locking would be expensive.

**Pull request closed without merging**